projects/azure/README.md

```diff
@@ -5,24 +5,184 @@ Main code repository is located in [Openshift Azure](https://github.com/openshif
+
+```
+ci-operator/templates/openshift/openshift-azure/
+```
+
+Obtain the `oc login` command from the OpenShift CI WebConsole and perform a login in your terminal
+
+Note: all further instructions assume you are currently in the root directory of this repository
 
-Populate secret file in `cluster/test-deploy/azure/secret` using `cluster/test-deploy/azure/secret_example` as template.
-If you are running this job not in CI cluster development namespace, but on other OpenShift cluster you will need to pre-populate base images in `azure` namespaces.
+Create a `secret` file for storing azure-related secrets using `cluster/test-deploy/azure/secret_example` as a template.
+
+```
+cp cluster/test-deploy/azure/secret_example cluster/test-deploy/azure/secret
+# insert/update credentials in cluster/test-deploy/azure/secret
+```
+
+Set the location of the namespace you wish to use for the ci-operator jobs
+```
+export CI_OPERATOR_NAMESPACE=your-chosen-namespace
+```
+
+#### Testing vm image builds
+
+Generate a new ssh keypair making sure to set the name of the keypair file as `ssh-privatekey`
+
+```
+ssh-keygen -t rsa -b 4096 -f cluster/test-deploy/azure/ssh-privatekey 
+```
+
+Checkout the repository containing the yum client certificates required by openshift-ansible to update the packages on the root VM.
+```
+git clone git@github.com:openshift/aos-ansible.git
+```
+
+Create `cluster/test-deploy/azure/certs.yaml` with the following format making sure to replace the `<contents of *>` placeholders
+```
+yum_client_cert_contents: |
+	<contents of aos-ansible/playbooks/roles/ops_mirror_bootstrap/files/client-cert.pem>
+
+yum_client_key_contents: |
+	<contents of aos-ansible/playbooks/roles/ops_mirror_bootstrap/files/client-key.pem>
+```
 
-Run job:
+Example: Build a rhel base vm image
 ```
-ci-operator --config ci-operator/config/openshift/openshift-azure/master.yaml --namespace=namespace-name --git-ref=openshift/openshift-azure@master --template ci-operator/templates/cluster-launch-e2e-azure.yaml --secret-dir $(pwd)/cluster/test-deploy/azure/
+docker run \
+--rm \
+-it \
+--env DEPLOY_OS=rhel7 \
+--volume $HOME/.kube/config:/root/.kube/config \
+--volume $(pwd):/release \
+registry.svc.ci.openshift.org/ci/ci-operator:latest \
+--config /release/ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml \
+--git-ref=openshift/openshift-azure@master \
+--namespace=${CI_OPERATOR_NAMESPACE} \
+--template /release/ci-operator/templates/openshift/openshift-azure/build-base-image.yaml \
+--secret-dir /release/cluster/test-deploy/azure/ \
+--target build-base-image
 ```
 
+Example: Build a centos base vm image
+```
+docker run \
+--rm \
+-it \
+--env DEPLOY_OS=centos \
+--volume $HOME/.kube/config:/root/.kube/config \
+--volume $(pwd):/release \
+registry.svc.ci.openshift.org/ci/ci-operator:latest \
+--config /release/ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml \
```

Comment:
You need to pass openshift-ansible as a config here although technically openshift-azure also works because the ansible image gets pulled in both cases, right? The problem with using openshift-azure is that you need to pass an extra variable in our template for the openshift release to use?

Reply:
I guess we need to pass OPENSHIFT_RELEASE anyway and today it's defaulted to 3.10. I am on the fence regarding the --config to use here but I feel the right approach is to use openshift-ansible.

/shrug

Reply:
we don't need `OPENSHIFT_RELEASE` for any of the `build-base-image`-templated jobs

Reply:
Yesm we don't need to pull openshift-azure either.